### ISSUE TITLE
F1 入力インテント解決関数（ジェスチャーライブラリ採用を最初から前提とする再発行）

### ISSUE THREAD

**Issue by rictaworks** [rictaworks/questboard#46]
## 依存
- #2

## 背景（旧Issue #10 / 旧Issue #44 の破棄と統合）

旧#10は「`InputIntentResolver`の意図解決ロジック」と「低レベルのポインタ/ジェスチャー集約実装」を分離せず一括発注していた。その結果PR #43（1次実装）は低レベルのポインタ状態機械を完全に自前実装し、レビューで以下5件のバグが見つかりクローズされた。

1. `pinchDistanceDeltaPx`が一度も算出されず、ピンチズームが常にパンとして扱われる
2. Space+左ドラッグがパンにならず、代わりにAlt+左ドラッグが誤ってパン扱いになる
3. `pointercancel`が通常の`pointerup`として処理され、選択状態が誤って変化する
4. `pointerdown`時に`setPointerCapture`していないため、キャンバス外へドラッグすると要素外でmove/upを取りこぼす
5. long-pressタイマーがデバイス/ボタン種別を無視するため、ペンの静止や右・中ボタン長押しで誤ってラジアルメニューが開く

事後的に「低レベル層のみライブラリへ移行する」Issue #44を発行したが、#44はPR #43のブランチから分岐することを前提にしており、PR #43がクローズされたことで前提が崩れ、実際に着手されたPR #45は（誤って）`main`から分岐され`input-intent-resolver.ts`自体が存在せず、実装コミットが一切積まれないまま空PRとなった。

→ 旧#10・旧#44は本Issueに統合しクローズする。**低レベルのポインタ/ジェスチャー集約は最初から自前実装せず、実績のあるライブラリを用いる**ことを最初から要件に含めることで、「自前実装→バグ→ライブラリへ移行」という二段階の手戻りを避ける。

## スコープ

デバイス種別・ボタン/タッチ数・修飾キー・ヒット対象・移動量・押下時間・現在選択・パーム接触面積を受け取り、優先順位に従って単一の意図を返す純粋関数/クラス（クラス図の`InputIntentResolver`）を実装し、キャンバス上のポインタ/ホイール/キーボードイベントに接続する。レンダリング方式はDOM/CSS+SVG採用（`app-ui/Questboard Prototype.dc.html`の`e.target.closest('[data-obj-id]')`方式のヒットテストを踏襲）。

**低レベルのポインタ/ジェスチャー集約（マルチポインタの追跡、ピンチ距離差分、pointer capture等）は自前実装せず、`@use-gesture/vanilla`（`@use-gesture/core`のフレームワーク非依存ラッパー、MIT、peer依存ゼロ）を用いて`CanvasInputController`を構築すること。** `InputIntentResolver`（意図解決の純粋ロジック、DOM非依存）と`CanvasInputController`（ライブラリ配線層）を最初から分離した設計とする。`InputIntentResolver`自体はDOM/イベントライブラリに依存しない純粋関数/クラスのまま維持する。

### `@use-gesture/vanilla`採用理由
- `react`版ではなく`vanilla`版を選ぶ理由：`CanvasInputController`はReactコンポーネントに未配線のプレーンなclass API（`attach(element)/detach()`）を想定しており、`react`版はReactフック（`useGesture`）前提のため不要な作り直しと`jsdom`+`@testing-library/react`導入を招く。`vanilla`なら`.attach()/.detach()`パターンをそのまま維持でき、テストも`jsdom`不要。

### 実装に先立つ重要な技術的発見（`@use-gesture/core@10.3.1`/`@use-gesture/vanilla@10.3.1`の実ソースを読んで確認済み。ドキュメントの記載だけでは分からない部分）

1. `SUPPORT`はモジュール読込時に`window`の有無で一度だけ計算される（`'onpointerdown' in window`等をチェック）。Node環境には`window`が無いため、素の状態では`PinchGesture`は**リスナーを一切バインドしない**（`pinchConfigResolver.device()`が`undefined`を返しbindが起きない。Pinchには`'mouse'`フォールバックが無い）。`DragGesture`は`'mouse'`にフォールバックするので動くが、その場合`pointerCapture`が`false`になり`sharedConfig.window`（未定義）に`change`リスナーを貼ろうとして例外になる。
   → **対策**: `@use-gesture`をimportする前に`globalThis.window`へ最小限のfeature-detection用スタブを設定する（DOM実装ではなく形状検知のみを満たす。jsdomは不要）:
   ```js
   globalThis.window = {
     document: { createElement() { return {}; } },
     onpointerdown: null,
     ontouchstart: null,
     navigator: { maxTouchPoints: 0 }
   };
   ```
2. `setPointerCapture`/`releasePointerCapture`/`hasPointerCapture`が`event.target`に対し呼ばれる（Pinch側の`pointerStart`はガード無しの無条件呼び出し、try/catchも無い）。テストで使うfakeターゲットにはこの3メソッドのno-opスタブが必須:
   ```js
   class FakeTarget extends EventTarget {
     setPointerCapture() {}
     releasePointerCapture() {}
     hasPointerCapture() { return false; }
   }
   ```
3. `Gesture`設定は`window`を`config.window`として明示的に上書き可能。fakeターゲット自身を`window`としても渡せば、down/move/up/cancel全イベントを単一のfake `EventTarget`にdispatchする現行の`.attach()`と同じ配線に保てる。
4. `event.buttons`（ビットマスク、`event.button`のインデックスではない）でドラッグ開始可否が決まり、デフォルトは左ボタンのみ（`config.pointerButtons`既定値`1`）。F1は中央/右ボタンドラッグ＝パンも扱うため、`pointer: { buttons: -1 }`でこのゲートを無効化し、ボタン種別の意味づけは`InputIntentResolver`に委ねる。テスト側の合成イベントにも`buttons`ビットマスクのスタンプが必須（`button`だけでは不十分。1=左, 2=右, 4=中央）。
5. `state.touches`（同時ポインタ数）、`state.elapsedTime`、`state.movement`、`state.last`、`state.first`がライブラリ側で正しく計算される。これにより手製の`activePointers` Map・`freshTouchMoves`・`isPinchLatched`・`pinchInitialDistance`・pointerup座標再同期処理は**構造的に不要になる**。
6. `PinchGesture`のピンチ距離差分は`state.movement`（スケール比）ではなく`state.da`（`[distance, angle]`の生値）から、`state.first`時点の値との差分として算出する必要がある。旧PR43のバグ1（`pinchDistanceDeltaPx`未算出）はここに起因していた。実装前に必ず実際の`state`オブジェクトの形状を確認すること。
7. `DragEngine`は2本目のポインタが増えても1本目の`onDrag`を自動停止しないため、「3本指はignore」「2本中1本離脱時に1本指move扱いしない」は`state.touches`を見る薄いゲートを`onDrag`グルー側に残す必要がある。

### 旧PR43で判明したバグの再発防止（ライブラリ採用により構造的に解消される想定だが、実装時に必ず検証すること）
- ピンチ距離差分（`pinchDistanceDeltaPx`相当）が`state.da`から正しく算出され、`resolve`に渡っていること
- Spaceキー押下状態はkeydown/keyupで管理して`spaceKey`として渡すこと（`altKey`をパン判定に誤流用しないこと）
- `pointercancel`では意図をemitせず、タイマー・状態のみリセットすること
- キャンバス外へのドラッグでも`move`/`up`を取りこぼさないこと（`pointer.capture:true`で解消される想定。要検証）
- long-pressタイマーはデバイス種別・ボタン種別をゲートし、ペンの静止や右・中ボタン長押しで誤発火しないこと

### 存置（`InputIntentResolver`本体・意図解決ロジック）
パーム拒否、長押しタイマー→ラジアルメニュー、Spaceキーpanモディファイア、右クリック/contextmenu抑制、`handleWheel`/`handleContextMenu`/`handleDblClick`など、use-gestureが扱わない領域は薄いパススルー/前段フィルタとして`CanvasInputController`に残す。`InputIntentResolver`クラス自体・`resolveHitTargetFromElement`・`resolveCanvasIntent`はDOMやジェスチャーライブラリに依存しない純粋ロジックとして設計する。

## ブランチ
`main`から新規ブランチを作成すること（旧PR #43・PR #45のブランチは使用しない。いずれも構造的な問題を抱えた実装のため再利用しない）。

## 受け入れ基準（仕様書1.5 F1より逐語転記）

1. ホイール＝ズーム（Ctrl+ホイール＝精密ズーム、Shift+ホイール＝横パン）
2. 中ボタンドラッグ／右ボタンドラッグ／Space+左ドラッグ／2本指ドラッグ＝パン（2本指の指間距離変化が閾値超ならピンチズーム）
3. 右クリック、または「押下500ms以上かつ移動8px未満」の長押し＝ラジアルメニュー起動
4. ハンドル上の左ドラッグ＝リサイズ/回転、接続点ドラッグ＝接続線作成
5. オブジェクト上：左クリック＝選択（Shift+クリック＝追加/除外選択）、左ドラッグ＝移動（Ctrl+ドラッグ＝複製移動）
6. 空白：左クリック＝選択解除、マウスの左ドラッグ＝範囲選択（タッチは投げ縄ツールをラジアルメニューで明示選択した時のみ範囲選択）、ダブルクリック＝付箋作成
7. テキスト可能オブジェクトのダブルクリック＝テキスト編集開始
8. ペン＝描画。接触面積が閾値超の接触はパームとして拒否
9. いずれにも該当しない入力は無視（例外を出さない）

- テストマトリクス「3デバイス×5対象×4修飾×6操作＝360（無効組合せ除外後288）」を100%合格でカバーする
- 状態遷移図6.1（Idle→Pressing→{RadialMenu,Dragging,Selected}→{Panning/Marquee/Moving/Resizing}→Inertia/Idle等）を実装する
- 旧PR43で指摘された5件のバグ（上記「旧PR43で判明したバグの再発防止」参照）が再発しないことをテストで担保する
- `npx tsc --noEmit`・`node --test test/*.test.mjs`が通る

## 対象ファイル
- `src/lib/input-intent-resolver.ts`（`InputIntentResolver` + `CanvasInputController`）
- `test/input-intent-resolver.test.mjs`
- `SPEC/F1-input-intent-resolution.md`
- `package.json` / `package-lock.json`（`@use-gesture/vanilla`追加）

## 参考
- `app-ui/Questboard Prototype.dc.html`の`onCanvasPointerDown/Move/Up`, `onCanvasContextMenu`, `onCanvasDblClick`, `onKeyDown`はロジック参考として移植可能。
- 旧PR #43（クローズ済み）: 自前実装で見つかった5件のバグの実例（再発防止の参考。コード自体は再利用しない）
- 旧Issue #44（クローズ済み）: `@use-gesture/vanilla`採用の技術調査（本Issueに全文引き継ぎ済み）
